naga/src/as_diagnostic_file_path.rs

```diff
@@ -0,0 +1,76 @@
+//! [`AsDiagnosticFilePath`] and its supporting items.
+
+use alloc::borrow::Cow;
+
+#[cfg(feature = "std")]
+use std::path::Path;
+
+#[cfg(not(feature = "std"))]
+use alloc::string::String;
+
+mod sealed {
+    pub trait Sealed {}
+}
+
+/// A trait that abstracts over types accepted for conversion to the most
+/// featureful path representation possible; that is:
+///
+/// - When `no_std` is active, this represents types can be converted to `Cow<'_, str>`.
```

Comment:
This is subtlly different - I think we should just list out the types here, as this sounds like it's `where for<'a> Cow<'a, str>: From<T>` or so

Reply:
Just pushed up 7a1a8434. If you think it's good, let's squash-and-merge!